src/main/java/org/folio/circulation/item/service/CirculationItemsService.java

```diff
@@ -0,0 +1,85 @@
+package org.folio.circulation.item.service;
+
+
+import java.time.LocalDateTime;
+import java.util.Objects;
+import java.util.UUID;
+
+import org.folio.circulation.item.domain.entity.Item;
+import org.folio.circulation.item.domain.mapper.CirculationItemsMapper;
+import org.folio.circulation.item.exception.IdMismatchException;
+import org.folio.circulation.item.exception.ResourceAlreadyExistException;
+import org.folio.circulation.item.repository.CirculationItemsRepository;
+import org.folio.rs.domain.dto.CirculationItem;
+import org.folio.spring.exception.NotFoundException;
+import org.springframework.stereotype.Service;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+
+@Service
+@RequiredArgsConstructor
+@Log4j2
+public class CirculationItemsService {
+
+  private final CirculationItemsRepository circulationItemsRepository;
+
+  private final CirculationItemsMapper circulationItemsMapper;
+
+  public CirculationItem getCirculationItemById(String id) {
+    var item = getCirculationItemEntityOrNull(id);
+    return circulationItemsMapper.mapEntityToDto(item);
+  }
+
+  private Item getCirculationItemEntityOrNull(String id) {
+    return circulationItemsRepository.findById(UUID.fromString(id))
+           .orElse(null);
+  }
+
+
+    public CirculationItem createCirculationItem(String circulationItemId, CirculationItem circulationItem) {
+    if(Objects.nonNull(circulationItem.getId()) &&
```

Comment:
I think we can have a method to check id mismatch and checkItemExist as the same code is used in both createCirculationItem and updateCirculationItem method

Reply:
checkIfItemExists is used in different contexts for both cases.
check id mismatch --> done